File: Example/PrebidDemoJava/src/main/java/org/prebid/mobile/javademo/activities/SettingsActivity.java
package org.prebid.mobile.javademo.activities;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;

import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import org.prebid.mobile.javademo.R;
import org.prebid.mobile.javademo.databinding.ActivitySettingsBinding;
import org.prebid.mobile.javademo.utils.Settings;

public class SettingsActivity extends AppCompatActivity {

    private final Settings settings = Settings.get();
    private ActivitySettingsBinding binding;

    public static Intent getIntent(Context context) {
        return new Intent(context, SettingsActivity.class);
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_settings);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle(getString(R.string.settings));
        }

        initViews();
    }

    @Override
    protected void onStart() {
        super.onStart();
        binding.etRefreshTime.addTextChangedListener(new TextWatcher() {
            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                checkAllFields();
            }

            @Override
            public void afterTextChanged(Editable s) {
            }

            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
            }
        });
    }

    private void initViews() {
        binding.etRefreshTime.setText(String.valueOf(settings.getRefreshTimeSeconds()));
    }

    private void checkAllFields() {
        hideError();

        String refreshTime = binding.etRefreshTime.getText().toString();
        int refreshTimeInt;
        try {
            refreshTimeInt = Integer.parseInt(refreshTime);
        } catch (Throwable any) {
            refreshTimeInt = 0;
        }
        if (refreshTimeInt < 30 || refreshTimeInt > 120) {
            showError("Refresh time must be in range from 30 to 120 seconds");
            settings.setRefreshTimeSeconds(30);
            return;
        } else {
            settings.setRefreshTimeSeconds(refreshTimeInt);
        }
    }

    private void showError(String error) {
        binding.tvError.setText(error);
    }

    private void hideError() {
        binding.tvError.setText("");
    }

}

File: Example/PrebidDemoKotlin/src/main/java/org/prebid/mobile/prebidkotlindemo/activities/SettingsActivity.kt
package org.prebid.mobile.prebidkotlindemo.activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.widget.addTextChangedListener
import androidx.databinding.DataBindingUtil
import org.prebid.mobile.prebidkotlindemo.R
import org.prebid.mobile.prebidkotlindemo.databinding.ActivitySettingsBinding
import org.prebid.mobile.prebidkotlindemo.utils.Settings

class SettingsActivity : AppCompatActivity() {

    companion object {
        fun getIntent(context: Context) = Intent(context, SettingsActivity::class.java)
    }

    private val settings = Settings.get()
    private lateinit var binding: ActivitySettingsBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_settings)
        supportActionBar?.title = getString(R.string.settings)

        initViews()
    }

    override fun onStart() {
        super.onStart()
        initListeners()
    }

    private fun initViews() {
        binding.etRefreshTime.setText(settings.refreshTimeSeconds.toString())
    }

    private fun initListeners() {
        binding.etRefreshTime.addTextChangedListener {
            checkAllFields()
        }
    }

    private fun checkAllFields() {
        hideError()
        binding.apply {
            val value = etRefreshTime.text?.toString()?.toIntOrNull() ?: 0
            if (value < 30 || value > 120) {
                showError("Refresh time must be in range from 30 to 120 seconds")
                settings.refreshTimeSeconds = 30
                return
            } else {
                settings.refreshTimeSeconds = value
            }
        }
    }

    private fun showError(error: String) {
        binding.tvError.text = error
    }

    private fun hideError() {
        binding.tvError.text = ""
    }

}